Python/bytecodes.c

```diff
@@ -5285,7 +5287,14 @@ dummy_func(
                 lltrace_resume_frame(frame);
             }
 #endif
+        // This is a little complicated...
```

Comment:
I don't see why this is needed. 
This is a normal dispatch, like any other.
It should tailcall in the tailcalling interpreter and jump in the normal interpreter.

Reply:
The problem is that we need it to both jump AND tailcall in the tailcalling interpreter. This is due to to the fact that the label is used also by this outer part before we enter the tail call https://github.com/python/cpython/blob/main/Python/ceval.c#L817.

So it's:
1. If outside, we need to jump.
2. If inside the tailcall interpreter, we need to tailcall.

Reply:
That's a symptom of the poor current design. The code for unwinding should be separate function, and then these gotos wouldn't be needed. Adding more code to work around this will take even more effort to pick apart later.
I really does look like we need to refactor the "throw" code first, as I suggested in the issue.

I have some (mostly) working code for that and a clear idea of what to do, so I can do it if you want.
Hopefully it won't delay this PR by more than a few days.

Reply:
> That's a symptom of the poor current design.

Of `PyEval_PyEval_EvalFrameDefault`, not your PR. In case that wasn't clear.

Reply:
Ok please go ahead and open the PR to refsctor the rest of pyeval.

Reply:
Factoring out the unwinding isn't necessary, we can just move a few labels to avoid bytecodes.c -> ceval.c jumps.
That way, there is no need for the tailcalling interpreter to jump back into the interpreter (in tier 1).
https://github.com/python/cpython/pull/129525